compiler/src/dmd/statementsem.d

```diff
@@ -989,7 +989,17 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
                          (tn.ty != tv.ty && tn.ty.isSomeChar && tv.ty.isSomeChar)) &&
                         !Type.tsize_t.implicitConvTo(tindex))
                     {
-                        deprecation(fs.loc, "foreach: loop index implicitly converted from `size_t` to `%s`",
+                        bool over = true; // overflow possible
```

Comment:
rename this to `couldOverflow`
over is the abbriviation for overload in the DMD codebase.